omniduct/duct.py

```diff
@@ -237,13 +237,13 @@ class Duct(with_metaclass(ProtocolRegisteringQuirkDocumentedABCMeta, object)):
     class Type(Enum):
         """
         The `Duct.Type` enum specifies all of the permissible values of
-        `Duct.DUCT_TYPE`.
+        `Duct.DUCT_TYPE`. Also determines the order in which ducts are loaded by DuctRegistry.
```

Comment:
this change is not necessary for snowflake but I ran into this issue during development... 

Specifically if during initialization a duct checks a prepared field (e.g. remote) it'll check all prepared fields including cache, and since cache WAS after db, this resulted in a key error.

Reply:
This sounds like a bug, since all Duct implementations should be lazily prepared. Which `Duct` subclass preparation was not lazy?